gtwrap/interface_parser/function.py

```diff
@@ -29,15 +29,29 @@ class Argument:
     void sayHello(/*`s` is the method argument with type `const string&`*/ const string& s);
     ```
     """
-    rule = ((Type.rule ^ TemplatedType.rule)("ctype") +
-            IDENT("name")).setParseAction(lambda t: Argument(t.ctype, t.name))
+    rule = ((Type.rule ^ TemplatedType.rule)("ctype") + IDENT("name") +
+            Optional(EQUAL + (DEFAULT_ARG ^ Type.rule ^ TemplatedType.rule) +
+                     Optional(LPAREN + RPAREN))("default")
```

Comment:
is this optional used anywhere?

Reply:
Yes it is. It is required to parse default constructors, e.g. `= NonlinearFactorGraph()` since `Type` will only parse `NonlinearFactorGraph`.

Added a comment.